src/go/k8s/pkg/resources/headless_service.go

```diff
@@ -118,3 +124,14 @@ func (r *HeadlessServiceResource) HeadlessServiceFQDN() string {
 		'.',
 		r.Key().Namespace)
 }
+
+func (r *HeadlessServiceResource) getAnnotation() map[string]string {
+	if !r.pandaCluster.Spec.ExternalConnectivity.Enabled {
+		return nil
+	}
+
+	return map[string]string{
+		externalDNSHostname:  r.pandaCluster.Spec.ExternalConnectivity.SubDomain,
```

Comment:
nit: maybe we should add to the subdomain api docs a link to the external dns feature we're using here for reference?

Reply:
Ok, added